src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -27,37 +34,63 @@ string IComputeSystemProvider.DefaultComputeSystemProperties
 
     public string Properties => throw new NotImplementedException();
 
-    public ComputeSystemProviderOperation SupportedOperations => throw new NotImplementedException();
+    // No create operation supported
+    public ComputeSystemProviderOperation SupportedOperations => 0x0;
 
     private bool IsValid(JsonElement jsonElement)
     {
         return jsonElement.ValueKind != JsonValueKind.Undefined;
     }
 
-    public IEnumerable<IComputeSystem> GetComputeSystemsAsync(IDeveloperId? developerId)
+    public async Task<IEnumerable<IComputeSystem>> GetComputeSystemsAsync(IDeveloperId? developerId)
     {
         var computeSystems = new List<IComputeSystem>();
+        _mgmtSvc.DevId = developerId;
+        var projectJSONs = await _mgmtSvc.GetAllProjectsAsJsonAsync();
 
-        var mgmtSvc = _host.Services.GetService<IDevBoxManagementService>();
-        if (mgmtSvc != null)
-        {
-            return computeSystems;
-        }
-        else
+        if (IsValid(projectJSONs))
         {
-            // ToDo: Remove throw and add to return object
-            Log.Logger()?.ReportError($"Error getting systems: Rest Service not configured");
-            throw new ArgumentException($"Rest Service needs to be configured.");
+            Log.Logger()?.ReportInfo($"Found {projectJSONs.EnumerateArray().Count()} projects");
+            foreach (var dataItem in projectJSONs.EnumerateArray())
+            {

```

Comment:
If an exception happens for one of the dataItem we will not process rest of projectJSONs. Same for the loop below with boxes.EnumerateArray(). And we won't even return an empty list. 
Would it be more robust to log an error and continue processing?

Reply:
Good catch. Will change.